Website/Packages/datax-pipeline/src/modules/flowDefinition/components/flowDefinitionPanel.jsx

```diff
@@ -733,11 +737,11 @@ class FlowDefinitionPanel extends React.Component {
     onResampleInput(kernelId) {
         const version = this.props.kernelVersion + 1;
         this.props.onUpdateKernelVersion(version);
-        this.props.onResampleInput(this.props.flow, kernelId, version);
+        this.props.onResampleInput(this.props.flow.name, this.props.flow.displayName, this.props.flow.owner, this.props.flow.input.properties.inputSchemaFile, this.props.flow.input.properties.normalizationSnippet, this.props.flow.input.properties.inputEventhubConnection, this.props.flow.input.properties.inputSubscriptionId, this.props.flow.input.properties.inputResourceGroup, this.props.flow.input.properties.inputEventhubName, this.props.flow.input.type, this.props.flow.resamplingInputDuration, kernelId, version);
```

Comment:
That s a lot of params.  Should we pass in the Flow object instead?

Reply:
This is an intentional change. It used to pass in the flow object perviously since all of this code was in the same package: datax-pipeline package. If we had continued with that design. that would mean that the datax-query package will have dependency on the datax-pipeline package. This PR creates a dependency for the datax-pipeline package on the datax-query package.  What you suggest will 1. create a circular dependency. 2. When another customer uses this package, they will need to have a dependency on the datax-pipeline package which we do not want. In fact, any of these params can be optional. I have currently made rules as optional since rules are very specific to data-accelerator

Reply:
Interesting.  So it is because Flow is defined in Pipeline.  Could Flow be moved too at some point into common components?

Reply:
We should not need to move flow into the datax-query package as that data to my understanding is heavily coupled to the implementation details of the datax-pipeline package. For example, you wouldn't want to need to modify the flow object defined in the datax-query package just to introduce new state/behavior of the datax-pipeline package. Also, for the other customer that you know I'm talking about, they don't care about flows at all.

Reply:
I do agree that there are a lot of params, Anu will be addressing this by formalizing a params object to wrap the min requirements to pass in the functions as a single arg instead of N args.

Reply:
Updated

Reply:
Updated.